src/cellflow/training/_trainer.py

```diff
@@ -103,6 +106,10 @@ def train(
         self.training_logs = {"loss": []}
         rng = jax.random.PRNGKey(0)
 
+        for callback in callbacks:
+            if isinstance(callback, PCADecodedMetrics2):
+                callback.add_validation_adata(self.validation_adata)
```

Comment:
I don't think this would be needed if `validation_adata` was an attribute of the callback, would it?